tox.ini

```diff
@@ -15,27 +15,45 @@ package = skip
 basepython = python3
 setenv =
     LC_ALL = en_US.utf-8
-passenv=
+passenv =
     PYTEST_ADDOPTS
     HYPOTHESIS_PROFILE
+deps =
```

Comment:
Don't these next two sections kind of defeat the purpose of having multiple test environments? The reason there's multiple test environments is to minimize and isolate the dependencies needed to run the commands. If we don't see value in doing that (I don't see much...), then why not share a single virtualenv among all the commands rather than creating a separate one per command?

Even so, I would maybe want to keep the integration test requirements separate as those can be longer and fatter installs and not needed for most contributors.

One other thing to keep in mind is if we unconditionally install all test and integration requirements for every command we run, we do run the possibility of accidentally importing a test-only dependency in our code and then not catching it until sbuild time.

Reply:
> Don't these next two sections kind of defeat the purpose of having multiple test environments?

Yeah

> If we don't see value in doing that (I don't see much...)

Agreed.

> then why not share a single virtualenv among all the commands rather than creating a separate one per command?

I agree with you, however [it doesn't look like this is supported under tox 4](https://github.com/tox-dev/tox/issues/2788#issuecomment-1367437092). It wouldn't be difficult to wire up a test runner with a shared virtual environment with a more flexible test runner [like nox](https://github.com/wntrblm/nox), which allows creating a test runner which reuses virtual environments.

> Even so, I would maybe want to keep the integration test requirements separate as those can be longer and fatter installs and not needed for most contributors.

How much time / space do you think that these dependencies add to the initial run? After the environment is set up, it shouldn't add extra time since they are cached. I'm fine with that if you'd prefer that, but personally I'd lean towards something that's simpler to manage, since after the initial install for a particular environment it doesn't cost any extra time - and I'd bet that the different in install time / space isn't significant but maybe I'm wrong.

Reply:
> I agree with you, however https://github.com/tox-dev/tox/issues/2788#issuecomment-1367437092.

Sad

> How much time / space do you think that these dependencies add to the initial run? After the environment is set up, it shouldn't add extra time since they are cached. I'm fine with that if you'd prefer that, but personally I'd lean towards something that's simpler to manage, since after the initial install for a particular environment it doesn't cost any extra time - and I'd bet that the different in install time / space isn't significant but maybe I'm wrong.

It's not strongly held, but I also don't see one extra set of dependencies (that includes the other set) such that we have 2 basic configuration will make the overall tox file all that more complex. Caching is nice, but unless I'm using tox-uv (which is still buggy for me on 3.8), my env still winds up getting recreated more often than I would think is necessary.

Reply:
@TheRealFalcon I dropped the extra dependencies, let me know what you think.